cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -916,15 +923,197 @@ subroutine basal_stress_coeff (nx_block, ny_block,         &
+           hwater      ! water depth at tracer location (m)
+      
+      real (kind=dbl_kind), dimension (nx_block,ny_block,ncat), intent(in) :: &
+           aicen,    & ! partial concentration for last thickness category in ITD
+           vicen       ! partial volume for last thickness category in ITD (m)
+      
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(inout) :: &
+           Tbu         ! seabed stress factor (N/m^2)
+
+! local variables 
+
+      integer (kind=int_kind) :: &
+           i, j, ij, ii, n
+
+      integer, parameter :: &
+           ncat_b = 100, &  ! number of bathymetry categories
+           ncat_i = 100     ! number of ice thickness categories (log-normal)
 
+      real (kind=dbl_kind), parameter :: &
+           max_depth = 50.0_dbl_kind, & ! initial range of log-normal distribution  
+           mu_s = 0.1_dbl_kind, &       ! friction coefficient
+           sigma_b = 2.5d0              ! Standard deviation of bathymetry
+
+      real (kind=dbl_kind), dimension(ncat_i) :: & ! log-normal for ice thickness
+           x_k, & ! center of thickness categories (m)
+           g_k, & ! probability density function (thickness, 1/m)
+           P_x    ! probability for each thickness category
+
+      real (kind=dbl_kind), dimension(ncat_b) :: & ! normal dist for bathymetry
+           y_n, & ! center of bathymetry categories (m)
+           b_n, & ! probability density function (bathymetry, 1/m)
+           P_y    ! probability for each bathymetry category        
+
+      real (kind=dbl_kind), dimension(ncat) :: &
+           vcat, acat
+
+      integer, dimension(ncat_b) :: &
+           tmp  ! Temporary vector tmp = merge(1,0,gt)
+      
+      logical, dimension (ncat_b) :: &
+           gt
+
+      real (kind=dbl_kind) :: wid_i, wid_b, mu_i, sigma_i, mu_b, m_i, v_i ! parameters for PDFs
+      real (kind=dbl_kind), dimension(ncat_i):: tb_tmp
+      real (kind=dbl_kind), dimension (nx_block,ny_block):: Tbt ! seabed stress factor at t point (N/m^2) 
+      real (kind=dbl_kind) :: atot, x_kmax, x997
+      real (kind=dbl_kind) :: cut, rhoi, rhow, gravit, pi, puny
+
+      character(len=*), parameter :: subname = '(seabed2_stress_coeff)'
+
+      call icepack_query_parameters(rhow_out=rhow, rhoi_out=rhoi)
+      call icepack_query_parameters(gravit_out=gravit)
+      call icepack_query_parameters(pi_out=pi)    
+      call icepack_query_parameters(puny_out=puny)
+
+      Tbt=c0
+
+      do ij = 1, icellt
+         i = indxti(ij)
+         j = indxtj(ij)
+
+         atot = sum(aicen(i,j,1:ncat))
+
+         if (atot > 0.05_dbl_kind .and. hwater(i,j) < max_depth) then
+
+            mu_b = hwater(i,j)           ! mean of PDF (normal dist) bathymetry
+            wid_i = max_depth/ncat_i     ! width of ice categories
+            wid_b = c2*c3*sigma_b/ncat_b ! width of bathymetry categories
+
+            x_k = (/( wid_i*(i*c1-0.5d0), i=1, ncat_i )/) 
```

Comment:
i*c1-0.5d0 is a little awkward.  I think you must be changing i to a real value by multiplying by c1; we usually use the intrinsic functions to do that.  Since this expression is reused in the next line (and maybe later?), you could make a local vector with these values.